sw-emulator/lib/periph/src/soc_reg.rs

```diff
@@ -328,6 +405,27 @@ impl Bus for SocRegistersInternal {
 
     fn poll(&mut self) {
         self.regs.borrow_mut().poll();
+
+        let mut regs = self.regs.borrow_mut();
+        if regs.mailbox.get_notif_irq() {
```

Comment:
One change I might consider making in the future is to put this filtering state inside some sort of Rc<SocIrqFilter>, and just have peripherals call a .set_irq(true) on it to dispatch interrupts.